janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLStoreManager.java

```diff
@@ -0,0 +1,421 @@
+                                        .grouped(2)
+                                        .toMap(array -> Tuple.of(array.get(0), Integer.parseInt(array.get(1)))))))
+                .toJavaMap();
+
+        final Session s = this.cluster.connect();
+        s.execute(createKeyspace(keyspaceName)
+                .ifNotExists()
+                .with()
+                .replication(replication));
+        return s;
+    }
+
+    ExecutorService getExecutorService() {
+        return this.executorService;
+    }
+
+    Session getSession() {
+        return this.session;
+    }
+
+    String getKeyspaceName() {
+        return this.keyspace;
+    }
+
+    Map<String, String> getCompressionOptions(final String name) throws BackendException {
+        final KeyspaceMetadata keyspaceMetadata = Option.of(this.cluster.getMetadata().getKeyspace(this.keyspace))
+                .getOrElseThrow(() -> new PermanentBackendException(String.format("Unknown keyspace '%s'", this.keyspace)));
+        return Option.of(keyspaceMetadata.getTable(name))
+                .map(tableMetadata -> tableMetadata.getOptions().getCompression())
+                .getOrElseThrow(() -> new PermanentBackendException(String.format("Unknown table '%s'", name)));
+    }
+
+    @Override
+    public void close() throws BackendException {
+        try {
+            this.session.close();
+        } finally {
+            try {
+                this.cluster.close();
+            } finally {
+                this.executorService.shutdownNow();
+            }
+        }
+    }
+
+    @Override
+    public String getName() {
+        return String.format("%s.%s", getClass().getSimpleName(), this.keyspace);
+    }
+
+    @Override
+    public Deployment getDeployment() {
+        return this.deployment;
+    }
+
+    @Override
+    public StoreFeatures getFeatures() {
+        return this.storeFeatures;
+    }
+
+    @Override
+    public KeyColumnValueStore openDatabase(final String name, final Container metaData) throws BackendException {
+        return this.openStores.computeIfAbsent(name, n -> new CQLKeyColumnValueStore(this, n, getStorageConfig(), () -> this.openStores.remove(n)));
+    }
+
+    @Override
+    public StoreTransaction beginTransaction(final BaseTransactionConfig config) throws BackendException {
+        return new CQLTransaction(config);
+    }
+
+    @Override
+    public void clearStorage() throws BackendException {
+        final Future<Seq<ResultSet>> result = Future.sequence(
+                this.executorService,
+                Iterator.ofAll(this.cluster.getMetadata().getKeyspace(this.keyspace).getTables())
+                        .map(table -> Future.fromJavaFuture(this.executorService, this.session.executeAsync(truncate(this.keyspace, table.getName())))));
+        result.await();
+        if (result.isFailure()) {
+            throw EXCEPTION_MAPPER.apply(result.getCause().get());
+        }
```

Comment:
Yeah, I've done that because dropKeyspace can hang according to reports around the cassandra forums. It's also more inline with what the other cassandra backends do as well. We are trying to fix the intermittent failures due to connection startup in startEmbedded and this is one one of those things that makes it go slow and cause some of the failures.